### PR TITLE
Using simplecrawler and modularized into a separate file

### PR DESCRIPTION
**Pull request by p1ho**
Sorry about the delay! I had other things needed taken of this last couple days.

But I finally got around to **[simplecrawler](https://www.npmjs.com/package/simplecrawler)**.
It's actually quite nice, but its caching mechanism seems to [have issue with the newest node engine](https://github.com/simplecrawler/simplecrawler/pull/447) atm, I also got the error message.

So, I'm using the old approach with **[flat-cache](https://www.npmjs.com/package/flat-cache)**, which works.

I also moved the whole thing into a separate file called [getUrlsFromCrawler.js](https://github.com/p1ho/site-validator-cli/blob/development/lib/getUrlsFromCrawler.js) in the lib folder.

I'll admit I'm not too experienced with Promises and Async/Await, the code works, but I'd super appreciate it if you could point out things I did that could be better.

Thanks!!

### CODE REVIEW

File: lib/getUrlsFromCrawler.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+'use strict'
+
+module.exports = async (url, cacheTime) => {
```
Comment (zrrrzzt) [inline lib/getUrlsFromCrawler.js]: You don't need async keyword here

File: lib/getUrlsFromCrawler.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  return new Promise(function (resolve, reject) {
```
Comment (zrrrzzt) [inline lib/getUrlsFromCrawler.js]: If you have any async/await code inside the promise you would need to change it to 
```JavaScript
return new Promise(async function (resolve, reject) {
  ...
}
```

**Review by zrrrzzt — APPROVED** [review]
Two minor comments, but still it would work.

File: lib/getUrlsFromCrawler.js (hunk 2 above)
Comment (p1ho) [reply lib/getUrlsFromCrawler.js]: Fixed

File: lib/getUrlsFromCrawler.js (hunk 1 above)
Comment (p1ho) [reply lib/getUrlsFromCrawler.js]: Fixed

**Comment by p1ho** [comment]
Fixed issues.

Also optimized caching mechanism so sitemap caches are stored separately.
This should help with performance if the cache base really starts to build up.
Because normalized url usually will contain invalid characters like ":" or "/", so I used hashing algorithm to create signatures for the urls.
SHA1(hex encoding) is chosen because it [seems to be the fastest](https://automationrhapsody.com/md5-sha-1-sha-256-sha-512-speed-performance/) and security is not of concern.

**Pull request merged**